### ISSUE TITLE
Compilation error after adding '@nuxtjs/composition-api/module', Nuxt Composition api

### ISSUE THREAD

**Issue by ajaykaju** [nuxt/nuxt.js#9666]
Currently i'm learning nuxt js , i decided to choose Composition API with TypeScript like i'm doing for vue 3. So i followed the instructions from https://composition-api.nuxtjs.org/getting-started/setup. After adding  '@nuxtjs/composition-api/module' to buildModules of nuxt.config.js.
After starting the server using npm run dev i'm getting a FATAL error. 

```
✖ Nuxt Fatal Error                                                                                  │
   │                                                                                                       │
   │   Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/vue-composition-api.esm.js' is not   │
   │   defined by "exports" in G:\Nuxt\mynuxt\node_modules\@vue\composition-apipackage.json  
```

tried with yarn and npm , also created new project but the problem is always there after adding '@nuxtjs/composition-api/module' .  I can;t understand what is the real issue. Everything is fine with optional api.

Versions : 

node : v14.7.0
npm: 6.14.7

"@nuxtjs/composition-api": "^0.26.0",
 "core-js": "^3.15.1",
 "nuxt": "^2.15.7"

 "@nuxt/types": "^2.15.7",
 "@nuxt/typescript-build": "^2.1.0"

### The entire error

```
 FATAL  Package subpath './dist/vue-composition-api.esm.js' is not defined by "exports" in G:\Nuxt\mynuxt\node_modules\@vue\composition-apipackage.json

  at applyExports (internal/modules/cjs/loader.js:523:9)
  at resolveExports (internal/modules/cjs/loader.js:539:23)
  at Function.Module._findPath (internal/modules/cjs/loader.js:671:31)
  at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1072:27)
  at Function.resolve (internal/modules/cjs/helpers.js:81:19)
  at Function.p [as resolve] (node_modules\jiti\dist\jiti.js:1:52953)
  at Resolver.resolveModule (node_modules\@nuxt\core\dist\core.js:263:28)
  at ModuleContainer.compositionApiModule2 (node_modules\@nuxtjs\composition-api\dist\module\index.js:152:22)
  at ModuleContainer.addModule (node_modules\@nuxt\core\dist\core.js:239:34)
  at node_modules\@nuxt\utils\dist\utils.js:639:43


   ╭───────────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                       │
   │   ✖ Nuxt Fatal Error                                                                                  │
   │                                                                                                       │
   │   Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/vue-composition-api.esm.js' is not   │
   │   defined by "exports" in G:\Nuxt\mynuxt\node_modules\@vue\composition-apipackage.json                │
   │                                                                                                       │
   ╰───────────────────────────────────────────────────────────────────────────────────────────────────────╯

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mynuxt@1.0.0 dev: `nuxt`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mynuxt@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Ajay\AppData\Roaming\npm-cache\_logs\2021-08-12T18_53_46_617Z-debug.log
```

**Comment by DerYeger** [comment]
I had the same issue a few days ago. Running `yarn upgrade` (or the equivalent for npm) fixed the issue for me.

**Comment by ajaykaju** [comment]
@DerYeger Thank you for your reply. But the version upgrades does not helped me. Tried with both yarn and npm upgrades. So i completely removed node and npm from my system and after a restart i installed node again from the site , 

### Versions
` node : v16.6.2` and `npm: 7.20.6`. 

then i created nuxt app with yarn (before i used npm ) and chose npm for package manager.  And added nuxt composition with `npm i @nuxtjs/composition-api`  and also added '@nuxtjs/composition-api/module' in buildModules.  This solved my issue.  

I don't know what was the real reason . Anyways thank you again ! :smiley:

**Comment by DerYeger** [comment]
I've had quite some issues with updating/upgrading dependencies as well, Nuxt can be weird in that regard 😅.

I'm glad it worked out in the end!

**Comment by lianulloa** [comment]
I had this same issue and neither using `yarn upgrade` nor doing a "phoenix" solved it for me

I made a highly hacky work around to make it work:

At node_modules/@nuxtjs/composition-api/dist/module/index.js inside `compositionApiModule2` function
I changed the declaration of `capiEntrypoint` from:

  `const capiEntrypoint = '@vue/composition-api/dist/vue-composition-api.esm.js'`
to
 `const capiEntrypoint = '@vue/composition-api'`

because @vue/composition-api exposes it as the main entry point.

I know this isn't a solution and I don't know if it would better if I've made a PR, but I don't know where it should be.
Also I just wanted it to work because I just want to be able to learn Nuxt.

Hope this can help you spot the problem
